crypto/hd/ethsecp256k1.go

```diff
@@ -0,0 +1,74 @@
+package hd
+
+import (
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcutil/hdkeychain"
+	bip39 "github.com/cosmos/go-bip39"
+
+	ethaccounts "github.com/ethereum/go-ethereum/accounts"
```

Comment:
Importing "github.com/ethereum/go-ethereum/accounts" brings in LGPL'd source files at compile time.

I don't have a strong opinion about this being safe or not in Apache licensed code.

My impression is that Geth developers think this is fine but just flagging it.

Reply:
Could we get this functionality from the MIT licensed library below?
https://github.com/INFURA/go-ethlibs

Reply:
the other thing I can think of is putting the LGPL functionality behind a build flag.

Reply:
IIRC we prelaunch had problems around this, and explicitly had to separate out code directories / not re-use geth code.

I forget the details of this cc @ebuchman , or if it was even well-advised at the time

Reply:
We already have go-ethereum code in `/crypto/keys/secp256k1/internal/secp256k1/` and we make an appropriate note there in the README file. Probably we should move that to the LICENSE file.